go/vt/vttablet/tabletserver/vstreamer/uvstreamer.go

```diff
@@ -432,6 +432,62 @@ func (uvs *uvstreamer) sendTestEvent(msg string) {
 	}
 }
 
+// waitForSource ensures that the source is able to stay within defined bounds for
+// its MVCC history list (trx rollback segment linked list for old versions of rows
+// that should be purged ASAP) and its replica lag (which will be -1 for non-replicas)
+// to help ensure that the vstream does not have an outsized harmful impact on the
+// source's ability to function normally.
```

Comment:
I think it is worth adding per-vstreamer metrics (`/debug/vars`), for diagnostics. Otherwise it will be difficult to understand delays/stalling in workflows. These metrics come to mind:

*  parameter settings for this tablet 
* current value of both history length and repl lag
* current wait time, if waiting in a rowstreamer 

wdyt?

Reply:
I like the idea of adding some metrics, for sure. I haven't done that yet so I'll look into it. Thanks!

Reply:
You can add new metrics similar to those here:

https://github.com/vitessio/vitess/blob/441d4bd051245ecdae61746240b6fdab60fc9c19/go/vt/vttablet/tabletserver/vstreamer/engine.go#L81-L92

Reply:
Resolving this after our discussion — to de-noise the PR a bit. Thanks again for the great suggestion and help!